linkerd/protocol/base-http/src/main/scala/io/buoyant/linkerd/http/ZipkinTracePropagator.scala

```diff
@@ -0,0 +1,94 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.twitter.finagle.{SimpleFilter, Stack}
+import com.twitter.finagle.buoyant.Sampler
+import com.twitter.finagle.http.util.StringUtil
+import com.twitter.finagle.tracing.{Flags, SpanId, TraceId}
+import com.twitter.util.Try
+import io.buoyant.linkerd.{TracePropagator, TracePropagatorInitializer}
+import io.buoyant.router.http.{HeadersLike, RequestLike}
+
+class ZipkinTracePropagator[Req, H: HeadersLike](implicit requestLike: RequestLike[Req, H]) extends TracePropagator[Req] {
+  /**
+   * Read the trace id from the request, if it has one.
+   */
+  override def traceId(req: Req): Option[TraceId] = {
+    var traceId = ZipkinTrace.get(requestLike.headers(req))
+
+    traceId
+  }
+
+  /**
+   * Return a sampler which decides if the given request should be sampled, based on properties
+   * of the request (zipkin or linkerd if zipkin not present).  If None is returned, the decision of whether to sample the request is deferred
+   * to the tracer.
+   */
+  override def sampler(req: Req): Option[Sampler] = {
+    var sampler = ZipkinTrace.getSampler(requestLike.headers(req)).map(Sampler(_))
+
```

Comment:
Similarly, shouldn't we be clearing the sample headers?

Reply:
Not sure, same question as above: why would you clear l5d-sample for a Zipkin propagator plugin which deals with x-b3- headers?

Reply:
Old trace context is now fully cleared before setting new trace context.